_posts/2018-01-30-quicli.md

```diff
@@ -0,0 +1,294 @@
+---
+title: quicli
+categories:
+- rust
+---
+I recently wrote a new Rust crate: [quicli].
+It gives you a way of quickly writing CLI programs in Rust.
+(Hence the name.)
+In this post,
+I want to present the ideas and underlying philosophies of this project.
+
+[quicli]: https://github.com/killercup/quicli
+
+## Some history
+
+I've been writing CLI tools in Rust for a while now.
+It really surprised me how this language
+that works really well for system programming
+quickly became my go-to solution for problems
+I'd previously write Shell or Ruby scripts for.
+
+I published the post
+["5 Tips for Writing Small CLI Tools in Rust"][cli-tips]
+at the end of August 2017
+that you may have read,
+where I describes some of my take-aways from that
+in the form of short tips.
+I've got some great feedback to that post,
+and there were people telling me
+that they seriously started looking into Rust
+after having read that article.
+That totally made my day:
+Someone took a chance at learning something new
+and becoming part of the great community that Rust has
+because of something I did!
+How cool is that?
+
+Ever since then,
+I wanted to condense this down into a small framework or library
+that you can use when you are just starting out with Rust
+as well as when you just want to write a quick tool.
+Sadly, like so often,
+I didn't have the time or concentration
+to sit down and really do this.
+And here's the good news:
+For some reason[^9e-talk],
+after all these months,
+I've now published a very early version.
+
+[cli-tips]: {% post_url 2017-08-31-rust-cli-tips %}
+
+[^9e-talk]: Okay, I admit, I promised to give a talk about Rust, and I wanted to present some nice, clean example code. (Of course, in the end I didn't show quicli in the talk.)
```

Comment:
Since you've made these two separate sentences rather than an aside you can remove the ( and )

Reply:
yeah, the create-react-app setup tools famously has this feature